src/sequence_batch_scheduler.cc

```diff
@@ -110,12 +108,143 @@ SequenceBatchScheduler::Create(
   // runner. This is at least 1 even if the model doesn't support
   // batching.
   const size_t model_batch_size = std::max(1, config.max_batch_size());
-  size_t seq_slot_cnt = model_batch_size;
+  sched->seq_slot_cnt_ = model_batch_size;
   if (config.sequence_batching().has_oldest()) {
-    seq_slot_cnt =
+    sched->seq_slot_cnt_ =
         config.sequence_batching().oldest().max_candidate_sequences();
   }
 
+  // Create batchers.
+  RETURN_IF_ERROR(sched->CreateBatchers(model->BackgroundInstances()));
+
+  // Create a reaper thread that watches for idle sequences. Run the
+  // reaper a lower priority.
+  SequenceBatchScheduler* raw = sched.release();
+
+  raw->reaper_thread_exit_ = false;
+  raw->reaper_thread_.reset(
+      new std::thread([raw]() { raw->ReaperThread(10 /* nice */); }));
+
+  scheduler->reset(raw);
+
+  return Status::Success;
+}
+
+Status
+SequenceBatchScheduler::Update()
+{
+  // Find added and removed instances for this update.
+  std::vector<std::shared_ptr<TritonModelInstance>> added_instances;
+  std::unordered_set<TritonModelInstance*> removed_instances;
+  InstancesDiff(&added_instances, &removed_instances);
```

Comment:
Why the scheduler needs to "re-"discover this information when model which invokes `Update()` already knows what is the changes in the instances

Reply:
"re-"discover is not needed. Updated: [Refactor model reuse code](https://github.com/triton-inference-server/core/pull/207/commits/38ac91008a5d46ba63ca4e4e5ca795d87fd9c723)